lib/network.py

```diff
@@ -1062,27 +1037,131 @@ def follow_chain(self, index):
     def get_local_height(self):
         return self.blockchain().height()
 
-    def synchronous_get(self, request, timeout=30):
+    def __wait_for(it):
+        """Wait for the result of calling lambda `it`."""
         q = queue.Queue()
-        self.send([request], q.put)
+        it(q.put)
         try:
-            r = q.get(True, timeout)
+            result = q.get(block=True, timeout=30)
         except queue.Empty:
             raise util.TimeoutException(_('Server did not answer'))
-        if r.get('error'):
-            raise Exception(r.get('error'))
-        return r.get('result')
 
-    def broadcast(self, tx, timeout=30):
-        tx_hash = tx.txid()
+        if result.get('error'):
+            raise Exception(result.get('error'))
+
+        return result.get('result')
+
+    def __with_default_synchronous_callback(invocation, callback):
+        """ Use this method if you want to make the network request
+        synchronous. """
+        if not callback:
+            return Network.__wait_for(invocation)
+
+        invocation(callback)
+
+    def request_header(self, interface, height):
+        self.queue_request('blockchain.block.get_header', [height], interface)
+        interface.request = height
+        interface.req_time = time.time()
+
+    def map_scripthash_to_address(self, callback):
+        def cb2(x):
+            x2 = x.copy()
+            p = x2.pop('params')
+            addr = self.h2addr[p[0]]
+            x2['params'] = [addr]
+            callback(x2)
+        return cb2
+
+    def subscribe_to_addresses(self, addresses, callback):
+        hash2address = {
+            bitcoin.address_to_scripthash(address): address
+            for address in addresses}
+        self.h2addr.update(hash2address)
+        msgs = [
+            ('blockchain.scripthash.subscribe', [x])
+            for x in hash2address.keys()]
+        self.send(msgs, self.map_scripthash_to_address(callback))
+
+    def request_address_history(self, address, callback):
+        h = bitcoin.address_to_scripthash(address)
+        self.h2addr.update({h: address})
+        self.send([('blockchain.scripthash.get_history', [h])], self.map_scripthash_to_address(callback))
+
+    # NOTE this method handles exceptions and a special edge case, counter to
+    # what the other ElectrumX methods do. This is unexpected.
+    def broadcast_transaction(self, transaction, callback=None):
+        command = 'blockchain.transaction.broadcast'
+        invocation = lambda c: self.send((command, [str(transaction)]), c)
+
+        if callback:
+            invocation(callback)
+
         try:
-            out = self.synchronous_get(('blockchain.transaction.broadcast', [str(tx)]), timeout)
+            out = Network.__wait_for(invocation)
```

Comment:
This is weird.  __wait_for is not a class method.

Reply:
No need for annotations here (and if I would I'd use `@staticmethod`):

```
class Sheep():
    def sound(self):
        Sheep.__default_sound()
        
    def __default_sound():
    	print("beh")
              
Sheep().sound()  # 'beh'
Sheep.__default_sound()  # AttributeError 
```
(a bit of a silly example of cource)